calcOutdatedMetrics.js

```diff
@@ -1,69 +1,80 @@
+      const behindPatch = [];
+      const behindMinor = [];
+      const behindMajor = [];
 
-  packageList?.forEach(package => {
-    const [name, currentVersion, wanted, latestVersion] = package;
-    const [currentMajor, currentMinor, currentPatch] = currentVersion;
-    const [latestMajor, latestMinor, latestPatch] = latestVersion;
+      packageList?.forEach((package) => {
+        const [name, currentVersion, wanted, latestVersion] = package;
+        const [currentMajor, currentMinor, currentPatch] = currentVersion
+          .split(".")
+          .map(parseInt);
+        const [latestMajor, latestMinor, latestPatch] = latestVersion
+          .split(".")
+          .map(parseInt);
 
-    if (currentMajor >= latestMajor) {
-      if (currentMinor >= latestMinor) {
-        if (currentPatch >= latestPatch) {
-          upToDate.push(package);
+        if (currentMajor >= latestMajor) {
+          if (currentMinor >= latestMinor) {
+            if (currentPatch >= latestPatch) {
+              throw new Error(`package is not outdated: ${name}`);
+            } else {
+              behindPatch.push(package);
+            }
+          } else {
+            behindMinor.push(package);
+          }
         } else {
-          behindPatch.push(package);
+          behindMajor.push(package);
         }
-      } else {
-        behindMinor.push(package);
-      }
-    } else {
-      behindMajor.push(package);
-    }
-  });
+      });
 
-  const numUpToDate = upToDate.length;
-  const numBehindPatch = behindPatch.length;
-  const numBehindMinor = behindMinor.length;
-  const numBehindMajor = behindMajor.length;
+      const numBehindPatch = behindPatch.length;
+      const numBehindMinor = behindMinor.length;
+      const numBehindMajor = behindMajor.length;
+      const numUpToDate =
+        numberOfPackages - numBehindMajor - numBehindMinor - numBehindPatch;
 
-  const percentUpToDate = (numUpToDate / numberOfPackages).toFixed(2);
-  const percentBehindPatch = (numBehindPatch / numberOfPackages).toFixed(2);
-  const percentBehindMinor = (numBehindMinor / numberOfPackages).toFixed(2);
-  const percentBehindMajor = (numBehindMajor / numberOfPackages).toFixed(2);
+      const percentUpToDate = (numUpToDate / numberOfPackages).toFixed(4);
+      const percentBehindPatch = (numBehindPatch / numberOfPackages).toFixed(4);
+      const percentBehindMinor = (numBehindMinor / numberOfPackages).toFixed(4);
+      const percentBehindMajor = (numBehindMajor / numberOfPackages).toFixed(4);
 
-  const results = {
-    numberOfPackages,
-    percentUpToDate,
-    percentBehindPatch,
-    percentBehindMinor,
-    percentBehindMajor,
-  };
-  process.stdout.write(JSON.stringify(results, null, 4));
+      const results = {
+        numberOfPackages,
+        percentUpToDate,
+        percentBehindPatch,
+        percentBehindMinor,
+        percentBehindMajor,
+      };
+      process.stdout.write(JSON.stringify(results, null, 4));
+    }
```

Comment:
overkill but interesting solution in t[his article](https://www.delftstack.com/howto/javascript/javascript-round-to-2-decimal-places/#:~:text=instead%20of%202.01%20.-,Use%20the%20Math.,ensure%20the%20number's%20accurate%20rounding.) for the rounding problem

their solution:
```javascript
function roundToTwo(num) {
    return +(Math.round(num + "e+2")  + "e-2");
}
console.log(roundToTwo(2.005));
```

ofc your solution is good, just adding for curiosities sake